diff -Naur obs-studio-27.2.4-OE1/deps/media-playback/media-playback/decode.c obs-studio-27.2.4/deps/media-playback/media-playback/decode.c
--- obs-studio-27.2.4-OE1/deps/media-playback/media-playback/decode.c	2022-03-28 07:29:23.000000000 +0800
+++ obs-studio-27.2.4/deps/media-playback/media-playback/decode.c	2025-02-03 20:16:14.533386985 +0800
@@ -190,8 +190,10 @@
 		d->in_frame = d->sw_frame;
 	}
 
-	if (d->codec->capabilities & CODEC_CAP_TRUNC)
+	#if LIBAVCODEC_VERSION_MAJOR < 60
+	 if (d->codec->capabilities & CODEC_CAP_TRUNC)
 		d->decoder->flags |= CODEC_FLAG_TRUNC;
+	#endif
 	return true;
 }
 
diff -Naur obs-studio-27.2.4-OE1/deps/media-playback/media-playback/decode.h obs-studio-27.2.4/deps/media-playback/media-playback/decode.h
--- obs-studio-27.2.4-OE1/deps/media-playback/media-playback/decode.h	2022-03-28 07:29:23.000000000 +0800
+++ obs-studio-27.2.4/deps/media-playback/media-playback/decode.h	2025-02-03 20:18:42.400057943 +0800
@@ -37,8 +37,10 @@
 #endif
 
 #if LIBAVCODEC_VERSION_MAJOR >= 58
+#if LIBAVCODEC_VERSION_MAJOR < 60
 #define CODEC_CAP_TRUNC AV_CODEC_CAP_TRUNCATED
 #define CODEC_FLAG_TRUNC AV_CODEC_FLAG_TRUNCATED
+#endif
 #else
 #define CODEC_CAP_TRUNC CODEC_CAP_TRUNCATED
 #define CODEC_FLAG_TRUNC CODEC_FLAG_TRUNCATED
diff -Naur obs-studio-27.2.4-OE1/libobs/obs-ffmpeg-compat.h obs-studio-27.2.4/libobs/obs-ffmpeg-compat.h
--- obs-studio-27.2.4-OE1/libobs/obs-ffmpeg-compat.h	2022-03-28 07:29:23.000000000 +0800
+++ obs-studio-27.2.4/libobs/obs-ffmpeg-compat.h	2025-02-03 20:20:30.816727778 +0800
@@ -23,8 +23,10 @@
 #endif
 
 #if LIBAVCODEC_VERSION_MAJOR >= 58
+#if LIBAVCODEC_VERSION_MAJOR < 60
 #define CODEC_CAP_TRUNC AV_CODEC_CAP_TRUNCATED
 #define CODEC_FLAG_TRUNC AV_CODEC_FLAG_TRUNCATED
+#endif
 #define INPUT_BUFFER_PADDING_SIZE AV_INPUT_BUFFER_PADDING_SIZE
 #else
 #define CODEC_CAP_TRUNC CODEC_CAP_TRUNCATED
diff -Naur obs-studio-27.2.4-OE1/plugins/obs-ffmpeg/obs-ffmpeg-compat.h obs-studio-27.2.4/plugins/obs-ffmpeg/obs-ffmpeg-compat.h
--- obs-studio-27.2.4-OE1/plugins/obs-ffmpeg/obs-ffmpeg-compat.h	2022-03-28 07:29:23.000000000 +0800
+++ obs-studio-27.2.4/plugins/obs-ffmpeg/obs-ffmpeg-compat.h	2025-02-03 20:22:11.676730705 +0800
@@ -27,8 +27,10 @@
 #endif
 
 #if LIBAVCODEC_VERSION_MAJOR >= 58
+#if LIBAVCODEC_VERSION_MAJOR < 60
 #define CODEC_CAP_TRUNC AV_CODEC_CAP_TRUNCATED
 #define CODEC_FLAG_TRUNC AV_CODEC_FLAG_TRUNCATED
+#endif
 #define CODEC_FLAG_GLOBAL_H AV_CODEC_FLAG_GLOBAL_HEADER
 #else
 #define CODEC_CAP_TRUNC CODEC_CAP_TRUNCATED
diff -Naur obs-studio-27.2.4-OE1/plugins/win-dshow/ffmpeg-decode.c obs-studio-27.2.4/plugins/win-dshow/ffmpeg-decode.c
--- obs-studio-27.2.4-OE1/plugins/win-dshow/ffmpeg-decode.c	2022-03-28 07:29:23.000000000 +0800
+++ obs-studio-27.2.4/plugins/win-dshow/ffmpeg-decode.c	2025-02-03 20:24:11.500067525 +0800
@@ -99,8 +99,10 @@
 		return ret;
 	}
 
-	if (decode->codec->capabilities & CODEC_CAP_TRUNC)
+	#if LIBAVCODEC_VERSION_MAJOR < 60
+	 if (decode->codec->capabilities & CODEC_CAP_TRUNC)
 		decode->decoder->flags |= CODEC_FLAG_TRUNC;
+	#endif
 
 	return 0;
 }
